cpp/src/parquet/encryption/encryption_internal.cc

```diff
@@ -599,7 +599,13 @@ static std::string ShortToBytesLe(int16_t input) {
 
 std::string CreateModuleAad(const std::string& file_aad, int8_t module_type,
                             int16_t row_group_ordinal, int16_t column_ordinal,
-                            int16_t page_ordinal) {
+                            int32_t page_ordinal) {
+  if (ARROW_PREDICT_FALSE(page_ordinal > std::numeric_limits<int16_t>::max())) {
```

Comment:
I'd suggest explicitly explain that it is the restriction by AAD suffix of page ordinal: https://github.com/apache/parquet-format/blob/master/Encryption.md#442-aad-suffix

Reply:
Okay. The standard seems to be ambiguous. The format doesn't says that `page cannot larger than i16`. But AAD requires it. I follow the implemention in parquet-mr: If the page not use AAD, larger than i16 is ok. Otherwise, we cannot use it.

By the way, seems compiler flag not use `no-narrowing`, which may cause some possible narrow-cast issue.

Reply:
@pitrou Whats your opinion? I just follow the Java implementions here.